### ISSUE TITLE
DateTimePicker: provide get methods for DatePicker and TimePicker

### ISSUE THREAD

**Issue by HerbertsVaadin** [vaadin/flow-components#4496]
### Describe your motivation

My main motivation is to be able to cleanly implement "default time" feature:
https://github.com/vaadin/web-components/issues/668

I can see there being some other use-cases like wanting to set a specific width for date picker or time picker through java API. 

### Describe the solution you'd like

```
DatePicker dp = dateTimePicker.getDatePicker();
TimePicker tp = dateTimePicker.getTimePicker()
```

### Describe alternatives you've considered

Alternatives is to use reflection, which is frowned upon.
Probably can use css + theme to style it.
Can use `dateTimePicker.getElement().executeJs(...)` to achieve the "default time" implementation, but it's messy. 

### Additional context

Would like for it to be back-ported into V14.

**Comment by rolfsmeds** [comment]
We might end up refactoring the DateTimePicker to _not_ be simply a combination of a DatePicker and TimePicker, so that it would be more like other input fields (this has been requested several times over the years, and would be more in line with the corresponding component in V8). Exposing the sub-components like this would not work well with that change.

So, I think it would be better to introduce APIs e.g. for listening to date and time changes separately, as well as for setting the date and time parts of the value separately.

Related issues: https://github.com/vaadin/web-components/issues/668 and #4501

**Comment by HerbertsVaadin** [comment]
If that's the plan, then sure, it would make sense to provide separate listeners and date / time setters.
Currently I see on web-components part that Serhii is working on replacing extended versions of time/date pickers for vaadin-date-time-picker, so it seems this major refactoring is not coming soon. [vaadin/web-components/issues/5250 ](https://github.com/vaadin/web-components/issues/5250)

**Comment by web-padawan** [comment]
> it seems this major refactoring is not coming soon. https://github.com/vaadin/web-components/issues/5250

Yes, for now we will keep using date and time pickers. But in theory we might reconsider that in the future.
